android/anc/src/main/java/org/smartregister/fhircore/anc/data/patient/PatientRepository.kt

```diff
@@ -321,16 +326,10 @@ constructor(
           filterByPatient(Observation.SUBJECT, patientId)
         }
       }
-    if (observations.isNotEmpty()) {
-      val listOfObservations =
-        observations.distinctBy { it.effectiveDateTimeType == DateTimeType(Date()) }
-      finalObservation =
-        if (listOfObservations.isNotEmpty())
-          listOfObservations.sortedBy { it.effectiveDateTimeType.value }.last()
-        else observations.sortedBy { it.effectiveDateTimeType.value }.last()
-    }
 
-    return finalObservation
+    return if (observations.isNotEmpty()) {
+      observations.sortedBy { it.effectiveDateTimeType.value }.last()
+    } else Observation()
```

Comment:
This return statement can be simplified to:

```kotlin
 return observations.maxByOrNull { it.effectiveDateTimeType.value } ?:  Observation()
```

Reply:
updated